src/compositor.rs

```diff
@@ -302,6 +302,17 @@ impl Compositor {
     /// Enters the wayland event loop. Won't return until the compositor is
     /// shut off
     pub fn run(self) {
+        self.run_with(|_|
+            unsafe {
+                ffi_dispatch!(WAYLAND_SERVER_HANDLE,
+                              wl_display_run,
+                              (*COMPOSITOR_PTR).display);
+            })
+    }
+
+    pub fn run_with<F>(self, runner: F)
```

Comment:
Can you add documentation on this function describing why it's necessary? e.g that it allows you to use the wayland event fd manually.